components/salsa-macros/src/query_group.rs

```diff
@@ -213,7 +213,15 @@ pub(crate) fn query_group(args: TokenStream, input: TokenStream) -> TokenStream
 
         query_fn_definitions.extend(quote! {
             fn #fn_name(&self, #(#key_names: #keys),*) -> #value {
-                salsa::plumbing::get_query_table::<#qt>(self).get((#(#key_names),*))
+                // Create a shim to force the code to be monomorphized in the
+                // query crate. Our experiments revealed that this makes a big
+                // difference in total compilation time in rust-analyzer, though
+                // it's not totally obvious why that should be.
+                fn __shim(db: &dyn #trait_name,  #(#key_names: #keys),*) -> #value {
+                    salsa::plumbing::get_query_table::<#qt>(db).get((#(#key_names),*))
+                }
```

Comment:
Will this work before polymorphisation work lands? I hope that rustc is not dumb enough to monomorphise non-generic functions, but still, one can not be sure... I guess, I'll just benchmak this.

Reply:
We also want the same for `input` queries I think (for both get & set), I saw then in `cargo llvm-lines` output as well (with not as high impact obviously).

Reply:
This should not require polymorphization, as this function is not generic (unlike a closure, a nested function item does not inherit generics from its context).

Reply:
But by all means we should test

Reply:
(Pushed a commit extending this to input queries)